ocaml/tests/test_xapi_vbd_helpers.ml

```diff
@@ -12,48 +12,44 @@
  * GNU Lesser General Public License for more details.
  *)
 
-open OUnit
-open Test_common
+module T = Test_common
 
 (* Helpers for testing Xapi_vbd_helpers.valid_operations *)
 
 let setup_test ~__context ~vdi_fun =
-  let _sm_ref = make_sm ~__context () in
-  let sr_ref = make_sr ~__context () in
-  let (_: API.ref_PBD) = make_pbd ~__context ~sR:sr_ref () in
-  let vm_ref = make_vm ~__context () in
+  let _sm_ref = T.make_sm ~__context () in
+  let sr_ref = T.make_sr ~__context () in
+  let (_: API.ref_PBD) = T.make_pbd ~__context ~sR:sr_ref () in
+  let vm_ref = T.make_vm ~__context () in
   let vdi_ref = vdi_fun sr_ref in
-  let vbd_ref = make_vbd ~vDI:vdi_ref ~vM:vm_ref ~__context () in
+  let vbd_ref = T.make_vbd ~vDI:vdi_ref ~vM:vm_ref ~__context () in
   let vbd_record = Db.VBD.get_record_internal ~__context ~self:vbd_ref in
   vbd_ref, vbd_record
 
-let my_cmp a b = match a,b with
-  | Some(code1, _), Some(code2, _) -> code1 = code2
-  | None, None -> true
-  | _	-> false
-
-let string_of_api_exn_opt = function
-  | None -> "None"
-  | Some (code, args) ->
-    Printf.sprintf "Some (%s, [%s])" code (String.concat "; " args)
-
-let run_assert_equal_with_vdi ~__context ?(cmp = my_cmp) ?(expensive_sharing_checks=true) ~vdi_fun op exc =
+let run_assert_equal_with_vdi ~__context msg ?(expensive_sharing_checks=true) ~vdi_fun op expected_error_if_any =
   let vbd_ref, vbd_record = setup_test ~__context ~vdi_fun in
-  assert_equal
-    ~cmp
-    ~printer:string_of_api_exn_opt
-    exc (Hashtbl.find (Xapi_vbd_helpers.valid_operations ~__context ~expensive_sharing_checks vbd_record vbd_ref) op)
+  let get_error_code_of op =
+    let valid_ops = (Xapi_vbd_helpers.valid_operations ~__context ~expensive_sharing_checks vbd_record vbd_ref) in
+    match Hashtbl.find valid_ops op with
+    | Some (code, _::_ ) -> Some code
+    | Some _ | None -> None
```

Comment:
I think this should be simplified to 
```ocaml
| Some (code, _ ) -> Some code | None -> None
```

Reply:
I don't think it would be specific to 2-element lists, rather it would skip out empty lists, but either way you're right that it needs changing.

Reply:
Yes sorry that's correct.

Reply:
That said, do you think I should remove the case that takes the first element of  a `pair option`? As it is, this is designed to accept `Some api_error_code,params`, not `Some individual_error`.

Reply:
Yeah, I think it's best to remove it.

Reply:
I forgot how typechecking works :P of course this only expects `(string * 'a) option`